internal/controllers/parser.go

```diff
@@ -295,7 +301,48 @@ func UpdateConfigFromAPIOpts(envoyConfiguration *config.EnvoyConfiguration, prox
 				rt.Action = routeRoute
 
 				routesToAddToVirtualHost = append(routesToAddToVirtualHost, rt)
+			case finalOpts.Auth != nil:
+				fmt.Println("-----")
+				fmt.Printf("finalOpts.Auth not nil - finalOpts.Auth=%+v\n", finalOpts.Auth)
 
+				if finalOpts.Auth.PathPrefix != nil {
+
+				}
+
+				// clusterName := "ext-authz-http-service:9002"
+				upstreamServiceHost := finalOpts.Auth.AuthUpstream.Host.Hostname
+				upstreamServicePort := finalOpts.Auth.AuthUpstream.Host.Port
+
+				// upstreamServiceHost := "ext-authz-http-service"
+				// var upstreamServicePort uint32 = 9002
+				clusterName := generateClusterName(upstreamServiceHost, upstreamServicePort)
+				// clusterName := upstreamServiceHost
+
+				if !envoyConfiguration.ClusterExist(clusterName) {
+					fmt.Printf("cluster does not exist - add: %s %s:%d\n", clusterName, upstreamServiceHost, upstreamServicePort)
+					envoyConfiguration.AddCluster(
+						clusterName,
+						upstreamServiceHost,
+						upstreamServicePort,
+					)
+				} else {
+					fmt.Printf("cluster exists - skip add: %s %s:%d\n", clusterName, upstreamServiceHost, upstreamServicePort)
+				}
+
+				// httpExternalAuthorizationFilter, err := config.NewHTTPExternalAuthorizationFilter("ext-authz-http-service", 9002)
+				httpExternalAuthorizationFilter, err := config.NewHTTPExternalAuthorizationFilter(
+					upstreamServiceHost,
+					upstreamServicePort,
+					clusterName,
+				)
+				httpConnectionManagerBuilder.AppendFilterHTTPExternalAuthorizationFilterToStart(httpExternalAuthorizationFilter)
+				if err != nil {
+					fmt.Println("finalOpts.Auth not nil - could not add append - err=%w", err)
```

Comment:
could you use the logging library the project is using instead of fmt printf?

Reply:
Resolved.

Removed `Printf`s etc. and debugging lines from code.